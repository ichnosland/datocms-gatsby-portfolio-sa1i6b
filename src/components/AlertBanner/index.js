/**
*
* AlertBanner
*
*/

import styled, { css } from 'styled-components';
import { lighten, darken } from 'polished';
import FlexBox from '../FlexBox';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BannerStyle = css`
  ${({ tag }) => tag && css`display: inline-flex;`};
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  text-align: center;
  margin: ${(props) => props.margin ? props.margin : '0'};
  border: ${(props) => props.border ? `1px solid ${props.theme.brand}` : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  ${({ actioncolor }) => actioncolor && css`
    border: 1px solid ${colore.actions[actioncolor]};
    background-color: ${lighten(0.4, colore.actions[actioncolor])};
    color: ${darken(0.2, colore.actions[actioncolor])};
    svg {
      fill: ${darken(0.2, colore.actions[actioncolor])};
    }
  `}
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const AlertBanner = styled(FlexBox)`
  ${BannerStyle}
  padding: ${(props) => props.padding ? props.padding : '0.6em  1.6em'};
  svg {
    fill: ${({ color, actioncolor }) => {
    if (color) return color;
    if (actioncolor) return actioncolor;

    return colore.ui.txt;
  }};
  }
`;

AlertBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const InfoBanner = styled(AlertBanner)`
  padding: ${(props) => props.padding ? props.padding : '0.6em  1em'};
  ${media.lt667`
    flex-direction: column;
    svg {
      margin: 10px 0;
    }
  `}
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '0.6em'};
  `}
`;

InfoBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const IconBanner = styled(FlexBox)`
  ${BannerStyle}
  align-self: stretch;
  align-items: center;
  padding: ${(props) => props.padding ? props.padding : '9px 16px'};
  div {
    margin-right: 12px;
    align-items: center;
    &:last-child {
      margin-right: 0;
    }
  }
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '7px 12px'};
  `}
`;

IconBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export default AlertBanner;
