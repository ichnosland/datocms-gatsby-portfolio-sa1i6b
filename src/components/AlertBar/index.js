/**
*
* AlertBar
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import styled, { css } from 'styled-components';

import Div from '../Div';
import FlexBox from '../FlexBox';
import Svg from '../Svg';
import { IconBtn } from '../TopBar';
import { H3 } from '../Heading';
import media from 'style/mediainjector';
import { colore } from 'style/color';
import icon from 'icons/globals';
import { ContainerStyle } from '../Container';

export const AlertWrap = styled(Div)`
  position: absolute;
  top: 0;
  left: 0;
  color: ${colore.ui.contrast};
  width: 100%;
  height: 56px;
  padding: 0 20px;
  background-color: ${(props) => props.theme.brand};
  ${({ giusto }) => giusto && css`background-color: ${colore.actions.okay};`}
  ${({ sbagliato }) => sbagliato && css`background-color: ${colore.actions.error};`}
  ${({ aiuto }) => aiuto && css`background-color: ${colore.actions.help};`}
  z-index: 8;
  transition: all 0.5s;
  ${media.desktop`
    border-top-left-radius: ${(props) => props.versione ? '3px' : '20px'};
    border-top-right-radius: ${(props) => props.versione ? '3px' : '20px'};
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

AlertWrap.defaulProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const BarContainer = styled(FlexBox)`
  ${ContainerStyle}
  height: 100%;
  ${(props) => props.helpFx && css`
    justify-content: space-between;
    &::after {
      content: "";
      display: inline;
    }
  `}
  h3 {
    color: ${colore.ui.contrast};
  }
`;

class AlertBar extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      giusto,
      sbagliato,
      aiuto,
      versione,
      show,
      helpFx,
    } = this.props;
    return (
      <AlertWrap
        giusto={giusto}
        sbagliato={sbagliato}
        aiuto={aiuto}
        show={show}
        versione={versione}
      >
        <BarContainer helpFx={helpFx}>
          {helpFx &&
            <IconBtn onClick={helpFx}>
              <Svg {...icon.topBarFeedback} />
            </IconBtn>
          }
          <H3>
            {giusto &&
              'Risposta corretta!'
            }
            {sbagliato &&
              'Hai sbagliato!'
            }
            {aiuto &&
              'Aiuto'
            }
          </H3>
        </BarContainer>
      </AlertWrap>
    );
  }
}

AlertBar.propTypes = {
  giusto: PropTypes.bool,
  sbagliato: PropTypes.bool,
  aiuto: PropTypes.bool,
  versione: PropTypes.bool,
  show: PropTypes.bool.isRequired,
  helpFx: PropTypes.func,
};

export default AlertBar;
