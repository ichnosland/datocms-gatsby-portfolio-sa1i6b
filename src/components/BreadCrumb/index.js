/**
*
* BreadCrumb
*
*/

import styled, { css } from 'styled-components';

import Div from 'components/Div';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BreadCrumb = styled(Div)`
  color: ${(props) => props.color};
  position: relative;
  display: ${(props) => props.display ? props.display : 'inline-block'};
  text-align: center;
  border: ${(props) => props.border ? props.border : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const Crumb = styled(Div)`
  position: relative;
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
  flex-wrap: nowrap;
  padding: 0.4em 1.4em 0.4em 0.8em;
  &:first-child {
    padding-left: 1.2em;
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:not(:first-child):not(:last-child) {
    &:after {
      content: "⟩";
      position: absolute;
      right: 0;
    }
  }
  &:only-child {
    &:after {
      content: "";
      position: relative;
    }
  }
`;

BreadCrumb.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export default BreadCrumb;
