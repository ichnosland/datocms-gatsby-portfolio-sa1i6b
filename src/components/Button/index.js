/**
*
* Button
*
*/

import styled, { css } from 'styled-components';
//import { lighten, darken, transparentize } from 'polished';
import media from 'style/mediainjector';
import { colore, transparent } from 'style/color';
import Svg from '../Svg';

export const ButtonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  font-family: ${(props) => props.theme.fonts.buttons.fontFamily};
  font-weight: ${(props) => props.theme.fonts.buttons.fontWeight};
  ${({ full }) => full && 'width: 100% !important;'};
  ${({ half }) => half && 'width: 50% !important;'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  text-align: center;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding ? props.padding : '0.4em  1.4em'};
  border: ${(props) => props.border ? props.border : '1px solid transparent'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : props.theme.brand};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.buttons};
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
    svg {
        opacity: .3;
      }
  }
  ${({ round }) => round && css`
    padding: 0;
    width: ${(props) => props.width};
    height: ${(props) => props.width};
    border-radius: 100%;
  `}
  ${({ actioncolor }) => actioncolor && css`
    background-color: ${colore.actions[actioncolor]};
    &:disabled {
      background-color: ${lighten(0.3, colore.actions[actioncolor])};
      color: ${lighten(0.1, colore.actions[actioncolor])};
    }
  `}
  ${({ admin }) => admin && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${colore.ui.neutralBg};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
  `}
  ${({ outline }) => outline && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${(props) => props.theme.brand};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
    svg {
      fill: ${(props) => props.fill ? props.fill : props.theme.brand};
    }
  `}
  ${({ bone }) => bone && css`
    color: ${colore.ui.contrast};
    border: 2px solid ${colore.ui.contrast};
    background-color: ${transparent};
    &:hover,
    &:active {
      background-color: rgba(0,0,0,.1);
    }
  `}
  &:active,
  &:focus {
    outline: none;
  }
  ${(props) => props.standard && css`
    min-width: 180px;
    max-width: 320px;
  `};
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${media.iphone45`
    ${({ standard }) => standard && 'min-width: 140px;'}
    padding: ${(props) => props.padding ? props.padding : '0.4em  0.6em;'};
    svg {
      margin: 0 6px 0 0;
    }
  `}
`;

export const Button = styled.button`
  ${ButtonStyle}
`;

Button.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const SubmitButton = styled.input`
  ${ButtonStyle}
`;

SubmitButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const ActionButtonStyle = css`
  font-size: 21px;
  padding: 21px 30px;
  background-color: ${(props) => props.bgcolor};
  border: ${(props) => props.border ? props.border : '2px solid white'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  font-weight: 600;
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${({ sectionbutton }) => sectionbutton && css`
    min-width: 50%;
    font-size: 18px;
    padding: 14px 30px;
    ${media.lt667`
      width: 100%;
    `}
  ${media.lt480`
    font-size: 16px;
    `}
  `}
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : colore.landing.green))};
    color: ${(props) => transparentize(0.5, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
  }
`;

export const ActionButton = styled(Button)`
  ${ActionButtonStyle};
`;

export const Icon = styled(Svg)`
  width: ${(props) => props.size ? props.size : '18px'};
  height: ${(props) => props.size ? props.size : '18px'};
  margin: ${(props) => props.margin};
  fill: ${(props) => props.fill ? props.fill : colore.ui.contrast};
  ${({ right }) => right && css`
    margin: 0 -6px 0 6px;
  `}
  ${({ left }) => left && css`
    margin: 0 6px 0 -6px;
  `}
  ${({ brandColor }) => brandColor && css`
    fill: ${(props) => props.theme.brand};
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

Icon.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const NoStyleButton = css`
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const GhostButton = styled.button`
  ${NoStyleButton}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  svg {
    fill: ${(props) => props.theme.brand};
  }
  ${(props) => props.fill && css`
    svg {
      fill: ${props.fill};
    }
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

GhostButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const TextButton = styled.button`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  appearance: none;
  border: none;
  background: none;
  &:active,
  &:focus {
    outline: none;
  }
  &:hover {
    ${media.desktop`
      color: ${(props) => props.theme.darken};
    `}
  }
`;

TextButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    darken: '#1588AA',
  },
};
