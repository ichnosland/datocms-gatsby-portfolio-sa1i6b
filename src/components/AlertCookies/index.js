/**
*
* AlertCookies
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import P from 'components/Paragraph';
import { Button } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const AlertCookiesWrap = styled(Div)`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 0 20px 20px;
  z-index: 1;
  ${media.lt667`
    padding: 0 10px 30px;
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

export const CookiesBanner = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  max-width: 860px;
  margin: 0 auto;
  font-size: 14px;
  border: ${(props) => props.border};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : 'rgba(255,255,255,.95)'};
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,.15);
  box-shadow: 0 2px 1px rgba(0,0,0,.25);
  p {
    padding: 20px;
    margin: 0;
    &:last-child {
      padding: 0 20px 10px;
      text-align: right;
      ${media.lt480`
        padding: 0 20px 20px;
        text-align: center;
      `}
    }
  }
  ${media.lt667`
    font-size: 12px;
  `}
`;

class AlertCookies extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { show, color, border, bgcolor, onClickFunction } = this.props;
    return (
      <AlertCookiesWrap show={show}>
        <CookiesBanner
          color={color}
          border={border}
          bgcolor={bgcolor}
        >
          <P>Al fine di offrire il miglior servizio possibile questo sito utilizza cookies tecnici e di terze parti per funzionalità quali l&apos;accesso tramite social network e/o la visualizzazione di media. Se non acconsenti all&apos;utilizzo dei cookie di terze parti, alcune di queste funzionalità potrebbero essere non disponibili. Chiudendo questo banner o proseguendo con la navigazione acconsentirai al loro impiego. Per saperne di pi&ugrave; o negare il consenso a tutti o ad alcuni cookie puoi consultare <a href="https://maieuticallabs.it/privacy" target="_blank"><span>l&apos;informativa sulla privacy</span></a>.</P>
          <P><Button actioncolor="hint" standard={1} onClick={onClickFunction}>Chiudi</Button></P>
        </CookiesBanner>
      </AlertCookiesWrap>
    );
  }
}

AlertCookies.propTypes = {
  color: PropTypes.string,
  border: PropTypes.string,
  bgcolor: PropTypes.string,
  show: PropTypes.bool.isRequired,
  onClickFunction: PropTypes.func,
};

export default AlertCookies;
