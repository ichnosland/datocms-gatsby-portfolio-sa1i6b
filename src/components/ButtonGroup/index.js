/**
*
* ButtonGroup
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { darken } from 'polished';

import media from 'style/mediainjector';
import FlexBox from 'components/FlexBox';
import { ButtonStyle, Icon } from 'components/Button';
import icon from 'icons/buttons';
import { colore } from 'style/color';

export const Group = styled(FlexBox)`
  text-align: center;
  margin: ${(props) => props.margin ? props.margin : '0 auto'};
  ${(props) => props.full && css`
    width: 100%;
  `}
  ${(props) => props.tabs && css`
     border-bottom: 2px solid ${props.theme.brand};
  `}
`;

Group.defaultProps = {
  theme: {
    brand: '#00BBEF',
    fonts: {
      buttons: {
        fontFamily: 'Quicksand',
        fontWeight: '400',
      },
    },
  },
};

export function active(activebgColor, activeColor) {
  return css`
    background-color: ${activebgColor || ((props) => props.theme.brand)};
    color: ${activeColor || colore.ui.contrast};
  `;
}

export const InnerButton = styled.button`
  ${ButtonStyle}
  position: relative;
  margin-left: -1px;
  color: ${(props) => props.color ? props.color : props.theme.brand};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  &:disabled {
    background-color: ${(props) => darken(0.1, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
    color: ${(props) => darken(0.2, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
    svg {
      fill: ${(props) => darken(0.2, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
    }
  }
  border: ${(props) => props.border ? props.border : '1px solid rgb(231, 236, 236)'};
  border-radius: 0;
  svg {
    fill: ${(props) => props.color ? props.color : props.theme.brand};
  }
  ${(props) => props.attivo && css`
    position:relative;
    svg {
      fill: ${colore.ui.contrast};
    }
  `}
  ${(props) => props.full && css`
    flex-grow: 1;
  `}
  ${(props) => props.attivo && active(props.activebgColor, props.activeColor)};
  &:first-child {
    border-radius: ${(props) => props.theme.radius.buttons} 0 0 ${(props) => props.theme.radius.buttons};
  }
  &:last-child {
    border-radius: 0 ${(props) => props.theme.radius.buttons} ${(props) => props.theme.radius.buttons} 0;
  }
  &:first-child:last-child {
    border-radius: ${(props) => props.theme.radius.buttons};
  }
  ${(props) => props.tabs && css`
    margin: 0 1px;
    border-bottom: none;
    border-radius: ${props.theme.radius.general} ${props.theme.radius.general} 0 0};
    &:first-child {
      border-radius: ${props.theme.radius.general} ${props.theme.radius.general} 0 0};
      margin-left: 0;
      margin-right: 1px;
    }
    &:last-child {
      border-radius: ${props.theme.radius.general} ${props.theme.radius.general} 0 0};
      margin-left: 1px;
      margin-right: 0;
    }
  `}
  &:active {
    ${(props) => active(props.activebgColor, props.activeColor)};
    svg {
      fill: ${colore.ui.contrast}
    }
  }
  ${media.lt667`
    padding: 0.4em  0.8em;
    ${({ hideOnMobile }) => hideOnMobile && 'display: none;'};
  `}
  ${media.iphone45`
    padding: 0.4em  0.6em;
  `}
`;

InnerButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
  },
};

class ButtonGroup extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      margin,
      buttons,
      half,
      full,
      color,
      bgcolor,
      border,
      activebgColor,
      activeColor,
      theme,
      tabs,
    } = this.props;
    return (
      <Group
        margin={margin}
        full={full}
        tabs={tabs}
        noPrint
      >
        {buttons.map((button) => (
          <InnerButton
            key={button.id}
            attivo={button.attivo}
            onClick={button.onClickFunction}
            half={half}
            full={full}
            color={color}
            bgcolor={bgcolor}
            border={border}
            activebgColor={activebgColor}
            activeColor={activeColor}
            theme={theme}
            hideOnMobile={button.hideOnMobile}
            disabled={button.disabled}
            tabs={tabs}
          >
            {(button.icona || []).length > 0 && <Icon {...icon[button.icona]} left />}
            {button.label}
          </InnerButton>
        ))}
      </Group>
    );
  }
}

ButtonGroup.propTypes = {
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      onClickFunction: PropTypes.func,
      id: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
      ]),
      attivo: PropTypes.bool,
    })
  ).isRequired,
  half: PropTypes.bool,
  full: PropTypes.bool,
  tabs: PropTypes.bool,
  color: PropTypes.string,
  margin: PropTypes.string,
  bgcolor: PropTypes.string,
  border: PropTypes.string,
  activebgColor: PropTypes.string,
  activeColor: PropTypes.string,
  theme: PropTypes.object,
};

export default ButtonGroup;
