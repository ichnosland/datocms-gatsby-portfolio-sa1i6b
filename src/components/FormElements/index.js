/**
*
* FormElements
*
*/

import styled, { css } from 'styled-components';

import Section from 'components/Section';
import Div from 'components/Div';
import { Button, Icon } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const BasicInputStyle = css`
  display: block;
  box-sizing: border-box;
  color: ${(props) => props.theme.brand};
  width: 100%;
  text-align: ${(props) => props.align ? props.align : 'left'};
  appearance: none;
  padding: 6px 13px;
  border: 1px solid ${(props) => props.theme.light};
  background-color: ${colore.ui.contrast};
  ${({ note }) => note && css`
    color: ${colore.ui.txt};
    background-color: ${colore.ui.note};
    border-color: ${colore.ui.noteBorder};
  `}
  &::placeholder {
    position: relative;
    color: ${(props) => props.theme.subtle};
    ${({ note }) => note && css`
      color: ${colore.ui.noteBorder};
    `}
    opacity: 1;
    overflow: visible;
  }
  &:active,
  &:focus {
    outline: none;
  }
`;

BasicInputStyle.defaultProps = {
  theme: {
    brand: '#00BBEF',
    subtle: '#B4E1FF',
    light: '#C9E4FF',
    radius: {
      searchBox: '100px',
    },
  },
};

export const FormSection = styled(Section)`
  height: 100%;
  flex-direction: column;
  justify-content: space-between;
  &::before {
    content: "";
    display: inline;
  }
`;

export const CardForm = styled.form`
  width: 280px;
  margin: 0 auto;
  text-align: center;
  ${media.lt480`
    width: 240px;
  `}
`;

export const FormElement = styled(Div)`
  position: relative;
  margin: ${(props) => props.margin ? props.margin : '0 auto 18px'};
  ${(props) => props.errore && css`
    margin-bottom: 9px;
  }`
}`;


export const FormMessage = styled.span`
  display: inline-block;
  width: 100%;
  font-size: 15px;
  color: ${(props) => props.color ? props.color : colore.actions.error};
  padding: 6px 9px;
  margin-top: 9px;
  border: 1px solid ${colore.ui.neutralBg};
  background-color: ${colore.ui.contrast};
  border-radius: ${(props) => props.theme.radius.general};
  ${(props) => props.landing && css`
    font-size: smaller;
    padding: 2px 12px;
    border-radius: 60px;
    margin-bottom: 3px;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
`;

FormMessage.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export const FacebookButton = styled(Button)`
  background-color: ${colore.social.facebook};
  border-radius: 3px;
`;

export const FacebookIcon = styled(Icon)`
  top: -2px;
  fill: #fff;
`;
