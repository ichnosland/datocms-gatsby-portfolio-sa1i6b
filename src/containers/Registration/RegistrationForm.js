/**
*
* RegistrationForm
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form/immutable';
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';

import { FACEBOOK } from 'configuration';
import {
  CardForm,
  FormElement,
  FacebookButton,
  FacebookIcon,
} from 'components/FormElements';
import renderedField from 'components/FormElements/Field';
import { LegalInfo, Separator, SeparatorText } from 'components/Paragraph';
import { Button } from 'components/Button';
import LogoPlatform from 'components/LogoPlatform';
import icon from 'icons/buttons';
import validate from './validator';

export const required = (value) => value ? undefined : 'Obbligatorio';

export class Registration extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { invalid, product, handleSubmit, responseFacebookFunction } = this.props;

    return (
      <CardForm onSubmit={handleSubmit}>
        <FormElement margin="0 auto 36px">
          <LogoPlatform
            product={product}
            formLogo
          />
        </FormElement>
        <FormElement>
          <FacebookLogin
            appId={FACEBOOK.appId}
            language={FACEBOOK.language}
            version={FACEBOOK.version}
            callback={responseFacebookFunction}
            isMobile
            disableMobileRedirect
            fields="name,email"
            scope="public_profile,email"
            render={/* istanbul ignore next */ (renderProps) => (
              <FacebookButton full onClick={renderProps.onClick} type="button">
                <FacebookIcon {...icon.facebook} left /> Registrati con Facebook
              </FacebookButton>
            )}
          />
        </FormElement>
        <Separator>
          <SeparatorText>oppure</SeparatorText>
        </Separator>
        <FormElement>
          <Field name="email" component={renderedField} type="email" label="Indirizzo email" validate={[required]} />
        </FormElement>
        <FormElement>
          <Field name="password" component={renderedField} type="password" label="Scegli una password" validate={[required]} />
        </FormElement>
        <FormElement>
          <Field name="password2" component={renderedField} type="password" label="Verifica la password" validate={[required]} />
        </FormElement>
        <FormElement>
          <LegalInfo>
            <Field
              name="privacy"
              id="privacy"
              component="input"
              type="checkbox"
              validate={[required]}
            />
            &nbsp;
            <label htmlFor="privacy">&nbsp; Ho letto ed accetto le <a href="https://maieuticallabs.it/privacy" target="_blank">normative sulla privacy</a></label>
          </LegalInfo>
        </FormElement>
        <FormElement>
          <Button actioncolor="okay" full type="submit" disabled={invalid}>Invia</Button>
        </FormElement>
      </CardForm>
    );
  }
}

Registration.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  invalid: PropTypes.bool,
  product: PropTypes.string,
  responseFacebookFunction: PropTypes.func.isRequired,
};

const RegistrationForm = reduxForm({
  form: 'registration',
  validate,
  asyncBlurFields: ['username'],
})(Registration);

export default RegistrationForm;
