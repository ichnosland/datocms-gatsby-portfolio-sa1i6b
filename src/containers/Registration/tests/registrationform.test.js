import React from 'react';
import { shallow } from 'enzyme';

import { CardForm } from 'components/FormElements';
import { Registration, required } from '../RegistrationForm';

const mockProps = {
  handleSubmit: () => { },
  product: 'product',
  invalid: false,
  responseFacebookFunction: () => { },
};

describe('<Registration />', () => {
  it('should render a CardForm', () => {
    const renderedComponent = shallow(
      <Registration {...mockProps} />
    );
    expect(renderedComponent.find(CardForm).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('Required', () => {
  it('should return undefined if there is a value', () => {
    expect(required('valore')).toBeUndefined();
  });
  it('should return `Required` if there isn\'t a value', () => {
    expect(required()).toBe('Obbligatorio');
  });
});
