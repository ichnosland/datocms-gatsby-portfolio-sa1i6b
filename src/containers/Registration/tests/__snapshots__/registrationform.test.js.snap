// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> should render a CardForm 1`] = `
<FormElements__CardForm
  onSubmit={[Function]}
>
  <FormElements__FormElement
    margin="0 auto 36px"
  >
    <LogoPlatform
      formLogo={true}
      product="product"
    />
  </FormElements__FormElement>
  <FormElements__FormElement>
    <t
      appId={476698685826665}
      authType=""
      callback={[Function]}
      cookie={false}
      disableMobileRedirect={true}
      fields="name,email"
      isMobile={true}
      language="it_IT"
      onFailure={null}
      redirectUri="http://localhost/"
      render={[Function]}
      responseType="code"
      returnScopes={false}
      scope="public_profile,email"
      state="facebookdirect"
      version={3.1}
      xfbml={false}
    />
  </FormElements__FormElement>
  <Paragraph__Separator
    theme={
      Object {
        "light": "#C9E4FF",
      }
    }
  >
    <Paragraph__SeparatorText>
      oppure
    </Paragraph__SeparatorText>
  </Paragraph__Separator>
  <FormElements__FormElement>
    <Field
      component={[Function]}
      label="Indirizzo email"
      name="email"
      type="email"
      validate={
        Array [
          [Function],
        ]
      }
    />
  </FormElements__FormElement>
  <FormElements__FormElement>
    <Field
      component={[Function]}
      label="Scegli una password"
      name="password"
      type="password"
      validate={
        Array [
          [Function],
        ]
      }
    />
  </FormElements__FormElement>
  <FormElements__FormElement>
    <Field
      component={[Function]}
      label="Verifica la password"
      name="password2"
      type="password"
      validate={
        Array [
          [Function],
        ]
      }
    />
  </FormElements__FormElement>
  <FormElements__FormElement>
    <Paragraph__LegalInfo>
      <Field
        component="input"
        id="privacy"
        name="privacy"
        type="checkbox"
        validate={
          Array [
            [Function],
          ]
        }
      />
       
      <label
        htmlFor="privacy"
      >
          Ho letto ed accetto le 
        <a
          href="https://maieuticallabs.it/privacy"
          target="_blank"
        >
          normative sulla privacy
        </a>
      </label>
    </Paragraph__LegalInfo>
  </FormElements__FormElement>
  <FormElements__FormElement>
    <Button
      actioncolor="okay"
      disabled={false}
      full={true}
      theme={
        Object {
          "brand": "#00BBEF",
          "fonts": Object {
            "buttons": Object {
              "fontFamily": "inherit",
              "fontWeight": "inherit",
            },
          },
          "radius": Object {
            "buttons": "6px",
          },
        }
      }
      type="submit"
    >
      Invia
    </Button>
  </FormElements__FormElement>
</FormElements__CardForm>
`;
